core/src/main/java/org/infinispan/remoting/inboundhandler/action/PendingTxAction.java

```diff
@@ -70,17 +70,11 @@ protected ActionStatus init(ActionState state) {
          keysToLock.removeAll(context.getLockedKeys());
       }
 
-      if (keysToLock.isEmpty()) {
-         //nothing to do. nobody else was able to update the state from checking, so no need to check the CAS
-         cas(InternalState.CHECKING, InternalState.READY);
-         return ActionStatus.READY;
-      }
```

Comment:
yes to avoid checking for pending again when it reaches the interceptors. however, I've made some changes to avoid creating a promise and I'm storing the NO_OP instance.